src/opm/parser/eclipse/EclipseState/Grid/EclipseGrid.cpp

```diff
@@ -238,37 +352,145 @@ namespace Opm {
             m_minpvMode = MinpvMode::ModeEnum::EclSTD;
         }
 
-
         if (deck.hasKeyword<ParserKeywords::MINPVFIL>()) {
             const auto& record = deck.getKeyword<ParserKeywords::MINPVFIL>( ).getRecord(0);
             const auto& item = record.getItem<ParserKeywords::MINPVFIL::VALUE>( );
             std::fill(m_minpvVector.begin(), m_minpvVector.end(), item.getSIDouble(0));
             m_minpvMode = MinpvMode::ModeEnum::OpmFIL;
         }
     }
+    
+    void EclipseGrid::initGridFromEGridFile(Opm::EclIO::EclFile& egridfile, std::string fileName){
+
+        if (!egridfile.hasKey("GRIDHEAD")) {
+            throw std::invalid_argument("file: " + fileName + " is not a valid egrid file, GRIDHEAD not found");
+        }
+
+        if (!egridfile.hasKey("COORD")) {
+            throw std::invalid_argument("file: " + fileName + " is not a valid egrid file, COORD not found");
+        }
+
+        if (!egridfile.hasKey("ZCORN")) {
+            throw std::invalid_argument("file: " + fileName + " is not a valid egrid file, ZCORN not found");
+        }
+
+        if (!egridfile.hasKey("GRIDUNIT")) {
+            throw std::invalid_argument("file: " + fileName + " is not a valid egrid file, ZCORN not found");
+        }
+
+        const std::vector<std::string>& gridunit = egridfile.get<std::string>("GRIDUNIT");
+        
+        const std::vector<int>& gridhead = egridfile.get<int>("GRIDHEAD");
+        const std::array<int, 3> dims = {gridhead[1], gridhead[2], gridhead[3]};
+
+        m_nx=dims[0];
+        m_ny=dims[1];
+        m_nz=dims[2];
+
+        const std::vector<float>& coord_f = egridfile.get<float>("COORD");
+        const std::vector<float>& zcorn_f = egridfile.get<float>("ZCORN");
+
+        m_coord.assign(coord_f.begin(), coord_f.end());
+        m_zcorn.assign(zcorn_f.begin(), zcorn_f.end());
+
+        if (gridunit[0] != "METRES") {
+
+            const auto length = ::Opm::UnitSystem::measure::length; 
+            
+            if (gridunit[0] == "FEET"){
+                Opm::UnitSystem units(Opm::UnitSystem::UnitType::UNIT_TYPE_FIELD );
+                units.to_si(length, m_coord);
+                units.to_si(length, m_zcorn);
+            } else if (gridunit[0] == "CM"){
+                Opm::UnitSystem units(Opm::UnitSystem::UnitType::UNIT_TYPE_LAB );
+                units.to_si(length, m_coord);
+                units.to_si(length, m_zcorn);
+            } else {
+                std::string message = "gridunit '" + gridunit[0] + "' doesn't correspong to a valid unit system";
+                throw std::invalid_argument(message);
+            }
+        }
+        
+        if ((egridfile.hasKey("ACTNUM")) && (m_useActnumFromGdfile)) {
+            const std::vector<int>& actnum  = egridfile.get<int>("ACTNUM");
+            resetACTNUM( actnum );
+        } 
+
+        if (egridfile.hasKey("MAPAXES")) {
+            const std::vector<float>& mapaxes_f = egridfile.get<float>("MAPAXES");
```

Comment:
I guess this should be inside a gridunit testing block.

Reply:
I guess this is related to unit conversion of mapaxes ? Is this comment still valid ?

Reply:
No - this comment does not apply any more.